### ISSUE TITLE
Simplify JavadocStyleCheck

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#7416]
JavadocStyleCheck currently does 4 things, which are duplicates of other checks.
https://github.com/checkstyle/checkstyle/blob/7c693ccc937109e4498c7756af0b5548348f42a7/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java#L326-L349

1) Ensure package-info.java has a Javadoc.
2) Checks the first sentence ends with a period and doesn't contain a specific fragement.
3) Checks that the javadoc isn't empty.
4) Checks that the tags have respective close for an open.

1 is a duplicate of MissingJavadocPackage. 2 is a duplicate of SummaryJavadoc but doesn't have a scope restriction. 3 is a duplicate of MissingJavadocMethod and MissingJavadocType. 4 is the only thing unique to this check.

I say we start new issues to remove the duplicate functionality, and provide the missing functionality to the other checks. This will help identify what is left for this check to be converted to an AbstractJavadocCheck.

**Comment by romani** [comment]
@rnveach ,

> 4. Checks that the tags have respective close for an open.
 4 is the only thing unique to this check.

I think that all antlr based checks do this, as it is requirement of proper parsing.
if this is true , we can simply remove this Check, and make migration note on what Check(configs) to use instead.

**Comment by rnveach** [comment]
@romani I assume you are referring to https://checkstyle.sourceforge.io/writingjavadocchecks.html#Tight-HTML_rules .

This violation is disabled by default. I don't know the validation the check does versus tight html to know if they are the same or not. I can definitely run regression to see if they are as another issue.

Are you ok with me to proceed with everything else and start issues for each one?

**Comment by romani** [comment]
yes , https://checkstyle.org/config_javadoc.html#SummaryJavadoc , property `violateExecutionOnNonTightHtml` is disabled by default. But we can ask users to activate it if they care.

> I don't know the validation JavadocStyleCheck does versus tight html to know if they are the same or not

please briefly take a look. It is kind does not matter .... we will not support custom parsing of HTML ... so if any diff exists it does not matter that much.

> Are you ok with me to proceed with everything else and start issues for each one?

if you prove that usage `violateExecutionOnNonTightHtml=true` is better/same we simply need to remove Check, I am not sure why might need separate issues.

**Comment by rnveach** [comment]
> 2 ... SummaryJavadoc (since 6.0) but doesn't have a scope restriction.

Discussed offline and we will not add this property to SummaryJavadoc. If you write a javadoc, you should add a summary to it no matter it's scope.

If tight html does same validation as current JavadocStyleCheck (4) we will just remove the check itself.

**Comment by rnveach** [comment]
@romani I updated the first post. Check does 5 things instead of 4. One of the extra things it does is check if there are any tags listed that are not allowed.
https://github.com/checkstyle/checkstyle/blob/7c693ccc937109e4498c7756af0b5548348f42a7/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java#L209-L218

As for tight html, it does not look exactly the same as this check.
Example: http://rveach.no-ip.org/checkstyle/regression/reports/288/checkstyle/index.html#A552
I think tight html is very strict about everything that opens is closed. JavadocStyleCheck allows some tags to be ignored, like `<p>`.
https://github.com/checkstyle/checkstyle/blob/7c693ccc937109e4498c7756af0b5548348f42a7/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java#L200-L202

Since the 5th validation of this check isn't covered by anything else that I see, I don't think we can completely remove the check unless you want to split 4 and 5 into their own, separate checks.

**Comment by romani** [comment]
JavadocStyle is removed from Checkstyle at 13.9.0, users should use SummaryJavadoc instead
Details:  https://github.com/checkstyle/checkstyle/issues/20610
